lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3206,6 +3207,11 @@ bool IsUnsafeRequest::evaluate(Evaluator &evaluator, Decl *decl) const {
   if (decl->getAttrs().hasAttribute<UnsafeAttr>())
     return true;
 
+  if (auto *alias = dyn_cast<TypeAliasDecl>(decl))
+    if (auto *nominal = alias->getUnderlyingType()->getAnyNominal())
```

Comment:
Would this work with two levels of typealiases, e.g.
```cpp
using SpanOfInt = std::span<int>;
using AlsoSpanOfInt = SpanOfInt;
```

Reply:
Good point! I should probably make this recursive because this would also not catch if the underlying type itself is an associated type of an unsafe type.

Reply:
Looks like this is not the right fix. I will talk to Doug.